framework/src/play/i18n/Messages.java

```diff
@@ -196,7 +196,7 @@ public static Properties all(String locale) {
             return defaults;
         Properties mergedMessages = new Properties();
         mergedMessages.putAll(defaults);
-        if (locale != null && locale.length() == 5 && locales.containsKey(locale.substring(0, 2))) {
+        if (locale.length() == 5 && locales.containsKey(locale.substring(0, 2))) {
```

Comment:
I think you should keep the test `locale != null` as locale is public it can be set to null outside the class